include/swift/AST/DiagnosticsSema.def

```diff
@@ -547,6 +547,9 @@ ERROR(invalid_member_type,none,
 ERROR(invalid_member_type_suggest,none,
       "%0 does not have a member type named %1; did you mean %2?",
       (Type, Identifier, Identifier))
+ERROR(invalid_member_reference,none,
+      "invalid reference to a member %0 of %1",
```

Comment:
The code looks pretty good to me now, but I think we can still improve the error message. How about `%0 %1 of type %2 is not a type` (where %0 is a DescriptiveDeclKind, so you don't just have to say "member")?